docs/types.rst

```diff
@@ -0,0 +1,58 @@
+Type Annotations
+================
+
+``attrs`` comes with first class support for type annotations for both Python 3.6 (:pep:`526`) and legacy syntax.
+
+On Python 3.6 and later, you can even drop the :func:`attr.ib`\ s if you're willing to annotate *all* attributes.
+That means that on modern Python versions, the declaration part of the example from the README can be simplified to:
+
+
+.. doctest::
+
+   >>> import attr
+   >>> import typing
+
+   >>> @attr.s(auto_attribs=True)
+   ... class SomeClass:
+   ...     a_number: int = 42
+   ...     list_of_numbers: typing.List[int] = attr.Factory(list)
+
+   >>> sc = SomeClass(1, [1, 2, 3])
+   >>> sc
+   SomeClass(a_number=1, list_of_numbers=[1, 2, 3])
+   >>> attr.fields(SomeClass).a_number.type
+   <class 'int'>
+
+You can still use :func:`attr.ib` for advanced features, but you don't have to.
+
+Please note that these types are *only metadata* that can be queried from the class and they aren't used for anything out of the box!
+
+
+mypy
+----
+
+While having a nice syntax for type metadata is great, it's even greater that `mypy <http://mypy-lang.org>`_ as of 0.570 ships with a dedicated ``attrs`` plugin which allows you to statically check your code.
+
+Imagine you add another line that tries to instantiate the defined class using ``SomeClass("23")``.
+Mypy will catch that error for you:
+
+.. code-block:: bash
+
+   $ mypy t.py
+   t.py:12: error: Argument 1 to "SomeClass" has incompatible type "str"; expected "int"
+
+This happens *without* running your code!
+
+And it also works with one of the two supported Python 2-style annotations.
```

Comment:
And what about when one uses the `type` argument to `attr.ib`? Will mypy also typecheck those?

Reply:
It basically says so literally in the paragraph you’ve quoted. I take it as an Signal that i should be clearer.

Reply:
Sorry, I see now the example has a line with a type argument, only read the one above which uses py2 comment-style type annotations and misunderstood. All good, thanks!